algosdk/atomic_transaction_composer.py

```diff
@@ -0,0 +1,541 @@
+            signer (TransactionSigner): signer that will sign the transactions
+            method_args (list[ABIValue | TransactionWithSigner], optional): list of arguments to be encoded
+                or transactions that immediate precede this method call
+            on_complete (OnComplete, optional): intEnum representing what app should do on completion
+                and if blank, it will default to a NoOp call
+            note (bytes, optional): arbitrary optional bytes
+            lease (byte[32], optional): specifies a lease, and no other transaction
+                with the same sender and lease can be confirmed in this
+                transaction's valid rounds
+            rekey_to (str, optional): additionally rekey the sender to this address
+
+        """
+        if self.status != AtomicTransactionComposerStatus.BUILDING:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposer must be in BUILDING state for a transaction to be added"
+            )
+        if len(self.txn_list) + method.get_txn_calls() > self.MAX_GROUP_SIZE:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposer cannot exceed MAX_GROUP_SIZE transactions"
+            )
+        if not method_args:
+            method_args = []
+        if len(method.args) != len(method_args):
+            raise error.AtomicTransactionComposerError(
+                "number of method arguments do not match the method signature"
+            )
+        if not isinstance(method, abi.method.Method):
+            raise error.AtomicTransactionComposerError(
+                "invalid Method object was passed into AtomicTransactionComposer"
+            )
+
+        app_args = []
+        foreign_accounts = []
+        foreign_assets = []
+        foreign_apps = []
+        # For more than 14 args, including the selector, compact them into a tuple
+        additional_args = []
+        additional_types = []
+        txn_list = []
+        # First app arg must be the selector of the method
+        app_args.append(method.get_selector())
+        # Iterate through the method arguments and either pack a transaction
+        # or encode a ABI value.
+        for i, arg in enumerate(method.args):
+            if arg.type in abi.method.TRANSACTION_ARGS:
+                if not isinstance(method_args[i], TransactionWithSigner):
+                    raise error.AtomicTransactionComposerError(
+                        "expected TransactionWithSigner as method argument, but received: {}".format(
+                            method_args[i]
+                        )
+                    )
+                txn_list.append(method_args[i])
+            else:
+                if arg.type in abi.method.FOREIGN_ARRAY_ARGS:
+                    current_type = abi.UintType(8)
+                    if arg.type == "account":
+                        current_arg = len(foreign_accounts)
```

Comment:
think this should be `len(foreign_accounts)+1` since the `sender` is implicitly the 0th

Reply:
Good point